pkg/generate/generator.go

```diff
@@ -131,15 +134,20 @@ func (g *Generator) GetCRDs() ([]*ackmodel.CRD, error) {
 				// It's a Status field...
 				continue
 			}
-			if fieldConfig.From == nil {
+			if fieldConfig.From == nil && len(fieldConfig.Type) == 0 {
```

Comment:
A couple things...

Since `Type` is an optional field, it's best to make it a pointer type, so `*string`. Then, here you can check for existence using `fieldConfig.Type == nil `

However, I'm actually a little confused why we wouldn't always look up the Input shape ref. The only function I would have thought would change would be the cleanGoType() function and maybe the GetTypeDefs() stuff that determines Go types for the underlying CRD fields.